fastapi/middleware/asyncexitstack.py

```diff
@@ -1,21 +1,27 @@
-from typing import Optional
+from typing import Awaitable, Callable, Optional
 
 from fastapi.concurrency import AsyncExitStack
-from starlette.types import ASGIApp, Receive, Scope, Send
+from starlette.requests import Request
+from starlette.responses import Response
 
 
 class AsyncExitStackMiddleware:
```

Comment:
I don't think this needs to be (or should be) a public class, in particular, this PR will be a "breaking change" for anyone using it directly, but I'm leaving it as it was to minimize change.

Reply:
Yep! Indeed, there's a lot of stuff that shouldn't be "public". I plan on making most of it private in a future release, wait to see who complains, and figure out what really needs to be public. But yep, for a future one.